task_5/Oleg.Mikhailov/src/main/java/task6/Binary.java

```diff
@@ -1,20 +1,22 @@
 package task6;
 
+import static jdk.nashorn.internal.objects.NativeArray.reverse;
+
 public class Binary {
     public static void main(String[] args) {
         int i = 0;
         int number =30;
-        int[] mas = new int[]{0,0,0,0,0};
+
+        String numb = "";
         while(i < 5) {
 
             int tmp = number / 2;
-            int bin = number - tmp*2;
+            int bin = number % 2;
             number=tmp;
-            mas[i] = bin;
+            numb = numb + bin;
             i++;
         }
-        for(i =4; i>=0;i--) {
-            System.out.print(mas[i]);
-        }
+        String b = new StringBuffer(numb).reverse().toString();
```

Comment:
The use of the StringBuffer is the worst here since all the methods of this class are synchronized and performance will be slower than it could be.
Look at the alternative - StringBuilder, it has the same functionality but its methods are not synchronized.
Even better, starting from JDK 8 you can use StringJoiner class in the same way